File: kotlin-examples/src/main/java/java/Conditions.java
package java;

import java.util.Locale;

public class Conditions {
    public Locale getDefaultLocale(String deliveryArea){
        if (deliveryArea.equals("germany") || deliveryArea.equals("austria") || deliveryArea.equals("switzerland")) {
            return Locale.GERMAN;
        }
        if (deliveryArea.equals("usa") || deliveryArea.equals("england") || deliveryArea.equals("australia")) {
            return Locale.ENGLISH;
        }
        throw new IllegalStateException("Unsupported deliveryArea " + deliveryArea);
    }
    //or via switch and fall-through
}

File: kotlin-examples/src/main/java/kotlin/Conditions.kt
package kotlin

import java.util.Locale

fun getDefaultLocale(deliveryArea: String): Locale {
    when (deliveryArea){
        "germany", "austria", "switzerland" -> return Locale.GERMAN
        "usa", "england", "australia" -> return Locale.ENGLISH
        else -> throw IllegalStateException("Unsupported deliveryArea $deliveryArea")
    }
}
// or shorter:
fun getDefaultLocale2(deliveryArea: String): Locale = when (deliveryArea){
    "germany", "austria", "switzerland" -> Locale.GERMAN
    "usa", "england", "australia" -> Locale.ENGLISH
    else -> throw IllegalStateException("Unsupported deliveryArea $deliveryArea")
}